.github/workflows/wasmcloud.yml

```diff
@@ -655,3 +669,44 @@ jobs:
         continue-on-error: ${{ github.repository_owner != 'wasmCloud' }}
         run: cargo publish --token ${{ secrets.CRATES_PUBLISH_TOKEN }}
         working-directory: ./crates/${{ matrix.crate }}
+
+  smart-release:
+    if: github.event_name == 'workflow_dispatch'
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: true
+      - uses: actions-rust-lang/setup-rust-toolchain@v1
+        with:
+          cache: false
+      - name: Install cmake for smart-release
+        run: sudo apt update && sudo apt install cmake -y
+
+      - name: install smart-release
+        env:
+          RUSTFLAGS: ''
+        run: cargo install cargo-smart-release --git https://github.com/brooksmtownsend/cargo-smart-release --branch feat/update-workspace-dependencies
+      - name: dry run release
+        if: ${{ !inputs.execute }}
+        run: |
+          git config --global user.email "automation@wasmcloud.com"
+          git config --global user.name "wasmCloud automation"
+          cargo smart-release --update-crates-index --no-publish --execute --no-changelog-preview --no-push ${{ inputs.crate }}
```

Comment:
NIT: maybe not worth doing, but we could decide the args to `smart-release` via a metadata step early on and then just *use* them from this step, enabling us to collapse dry run and regular release steps?

Could probably factor out the `git config` lines too

Reply:
I also would like this but not required

Reply:
I definitely think there are improvements here but can tackle as followups